### PR TITLE
Rbac and autoschema

### PR DESCRIPTION
**Pull request by dirkkul**
### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [ ] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

File: test/acceptance/authz/autoschema_test.go [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+	adminKey := "admin-key"
+	adminUser := "admin-user"
+	adminAuth := helper.CreateAuth(adminKey)
+
+	readSchemaAction := authorization.ReadCollections
+	createDataAction := authorization.CreateData
+	updateSchemaAction := authorization.UpdateCollections
+	createSchemaAction := authorization.CreateCollections
+
+	_, down := composeUp(t, map[string]string{adminUser: adminKey}, map[string]string{customUser: customKey}, nil)
```
Comment (tsmith023) [inline test/acceptance/authz/autoschema_test.go:39]: This function doesn't append the `WithAutoSchema` builder that you've introduced so is this test actually running with auto schema? Maybe auto schema is on by default if not set?

File: usecases/objects/auto_schema.go [hunk 2]
```diff
@@ -72,37 +76,47 @@ func (m *autoSchemaManager) autoSchema(ctx context.Context, principal *models.Pr
...
 		if len(object.Class) == 0 {
 			// stop performing auto schema
-			return 0, fmt.Errorf(validation.ErrorMissingClass)
+			return 0, ErrInvalidUserInput{validation.ErrorMissingClass}
 		}
 
 		object.Class = schema.UppercaseClassName(object.Class)
 
-		vclass := vclasses[schema.UppercaseClassName(object.Class)]
+		vclass := vclasses[object.Class]
```
Comment (tsmith023) [inline usecases/objects/auto_schema.go:101]: Is this because the class name was capitalised further up in the call stack?

Comment (dirkkul) [reply usecases/objects/auto_schema.go:101]: directly the line above

Comment (tsmith023) [reply usecases/objects/auto_schema.go:101]: hahahaha, my brain

**Review by tsmith023 — APPROVED** [review]

**Pull request merged**